pkg/cri/server/container_create.go

```diff
@@ -332,6 +333,9 @@ func (c *criService) volumeMounts(containerRootDir string, criMounts []*runtime.
 		}
 		volumeID := util.GenerateID()
 		src := filepath.Join(containerRootDir, "volumes", volumeID)
+		if !filepath.IsAbs(dst) && goruntime.GOOS != "windows" {
```

Comment:
The volumes returned by this function, end up being passed to [`containerSpec()`](https://github.com/containerd/containerd/blob/main/pkg/cri/server/container_create.go#L159), which is platform specific. The better place to handle the path would be closer to where it actually ends up being used. This way each OS can handle it in accordance to the rules of that OS.

Reply:
Since the `cri/server` doesn't use sandbox API, we don't know the runtime platform actually, if I don't miss something here. So, I think we can keep it as it is here. Use host platform as condition. And it's easy to backport and maintain in LTS.

Reply:
That's true. My thought here was that since this ends up being used in platform specific files [`container_create_linux.go`](https://github.com/containerd/containerd/blob/main/pkg/cri/server/container_create_linux.go#L117), [`container_create_windows.go`](https://github.com/containerd/containerd/blob/main/pkg/cri/server/container_create_windows.go#L41), etc, it would spare us the extra check on `runtime.GOOS`.

Either way is fine with me.

Reply:
I'll let it like this as @fuweid and Derek also both said they prefer to just change it here, rather than in those functions that can affect mounts created by other means.

Thanks!

Reply:
nod I wasn't pointing over to the mount stuff to say call that I meant a new one local to both observer and server.. 

platform is just a type alias for oci platform.. and from what I see so far pod sandbox is ...
```
func (c *Controller) Platform(_ctx context.Context, _sandboxID string) (platforms.Platform, error) {
	return platforms.DefaultSpec(), nil
}
func DefaultSpec() specs.Platform {
	return specs.Platform{
		OS:           runtime.GOOS,
		Architecture: runtime.GOARCH,
		// The Variant field will be empty if arch != ARM.
		Variant: cpuVariant(),
	}
} 
```
you could just pass platform.OS from sbserver and pass runtime.GOOS from the other for now.. if we had a common util func.. The only reason really to suggest common was in case we need to use that common in other locations in server/sbserver code and not have to remember this pattern again :-)